src/test-utils/matchers.ts

```diff
@@ -0,0 +1,32 @@
+// This type is not exported from Jest, so we need to infer it from the expect.extend function.
+type MatcherFunction = Parameters<typeof expect.extend>[0] extends { [key: string]: infer I } ? I : never
+
+const isElementVisible = (element: HTMLElement): boolean => {
+  return element.style.height !== '0px' && (!element.parentElement || isElementVisible(element.parentElement))
+}
+
+// Overrides the Testing Library matcher to check for height when determining whether an element is visible.
+// We are doing this because:
+// - original `toBeVisible()` does not take `height` into account
+// https://github.com/testing-library/jest-dom/issues/450
+// - original `toBeVisible()` and `toHaveStyle()` does not work at all in some cases
+// https://github.com/testing-library/jest-dom/issues/209
+// - `getComputedStyles()` returns empty object, making it impossible to check for Styled Components styles
+// https://github.com/styled-components/styled-components/issues/3262
+// https://github.com/jsdom/jsdom/issues/2986
+// For the reasons above, this matcher only works for inline styles.
```

Comment:
This should be a doc-comment, not an implementation comment.
In other words, `/**` :P

Reply:
Why should it be a doc comment? I didn't know `//` was an implementation comment. So far, I think we made a distinction that we're only using `/**` where we plan to use JSDoc. I don't think applies here?

Reply:
It explains what a function is doing, so it's a doc-comment. In general, if you're commenting above a function, it's a doc-comment; if you're commenting in a function definition, it's an impl comment.

> So far, I think we made a distinction that /** was only to be used when JSDoc/TSDoc were present.

Using `/**` makes it JSDoc, even if you don't use specific JSDoc tags. With TypeScript, most JSDoc comments are very simple, just `/** Return xyz. */`, because the actual types are already documented through the language. If it's something you'd expect to be surfaced as documentation, explaining a function's usage, showing up through the IDE, it's a doc-comment and should utilize `/**`.

Reply:
You are right, I remember we made a distinction between function-level documents and multi-line comments inside the body (implementation). I will follow-up with an update.